src/cycles/johnson.jl

```diff
@@ -73,11 +73,11 @@ vertices to unblock if the key vertex is unblocked.
 """
 struct JohnsonVisitor{T <: Integer} <: Visitor{T}
     stack::Vector{T}
-    blocked::BitArray
+    blocked::BitArray{1}
     blockedmap::Vector{Set{T}}
 end
 
-@traitfn function JohnsonVisitor(dg::::IsDirected)
+function JohnsonVisitor(dg)

```

Comment:
why are we taking the traitfn away? AFAIU, it's still limited to digraphs....

Reply:
It's been a while since I have worked on this, but if I remember correctly the compiler was struggling in a lot of functions in this file with inferring the correct type - the problem seems to be the use of `Channel` (we might have to replace them in the future by iterators). So if I remember correctly, removing the `traitfn` made the situation slightly better.
Because `JohnsonVisitor` is not exported anyways, I thought it would not really hurt to remove the restriction.

Reply:
No worries. Thanks for the explanation. We should document this in the code though.